### PR TITLE
Display category and date of the pages in list and content fragments

### LINKED ISSUE #323: [list, content] Display date and category
**Is this a BUG REPORT or FEATURE REQUEST?**:
bug

**What happened**:
List and content fragment don't display content fragment's date.

**What you expected to happen**:
To see when the content was published in pages where it's required, but not always.

**How to reproduce it (as minimally and precisely as possible)**:
Create a content fragment. The date variable is not shown in the website.

### LINKED ISSUE #332: Move blog example into dev or fragments
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
Currently `/blog/` is used, but not linked.

**What you expected to happen**:
Either move `/blog/` into `/dev/` or maybe use it as a showcase item under `/fragments/` or `/showcase/`. With showcase we could showcase various usage examples of the fragments. One being the blog use case using list + content pages.

### PR DESCRIPTION
**Pull request by mpourismaiel**
**What this PR does / why we need it**:
Display date and categories in both content and list fragment

**Which issue this PR fixes**:
fixes #323 
fixes #332

**Release note**:
```release-note
- list: Display date and category of pages
- content: Display last modification date
```

### CODE REVIEW

File: layouts/partials/helpers/publish-date.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<span class="text-muted mr-2">
+  {{- if ne $.root.Params.Lastmod $.root.Params.Date -}}
```
[the remark this answers is not in the document]
Comment (mpourismaiel) [reply layouts/partials/helpers/publish-date.html]: I seriously have no idea. I copied over the code and was working on it for 15 minutes to get it to work :D I'll see if it has any effect.

File: layouts/partials/helpers/publish-date.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    {{- i18n "last_updated" | default "Last updated:" -}}
+  {{- end -}}
+  <time datetime="{{ $.root.Date }}" itemprop="datePublished dateModified">
+    {{- $.root.Lastmod.Format ($.site.Params.date_format | default "2006 January 26") -}}
```
[the remark this answers is not in the document]
Comment (stp-ip) [reply layouts/partials/helpers/publish-date.html]: That might actually be a weird Go Template issue so. But yeah weird.

File: layouts/partials/helpers/publish-date.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<span class="text-muted mr-2">
+  {{- if ne $.root.Params.Lastmod $.root.Params.Date -}}
+    {{- i18n "last_updated" | default "Last updated:" -}}
```
[the remark this answers is not in the document]
Comment (mpourismaiel) [reply layouts/partials/helpers/publish-date.html]: I removed this section entirely. With the new code, `.Lastmod` is used by default and if there is `date` available in frontmatter, that is used instead.

**Comment by mpourismaiel** [comment]
Extracted date to it's own partial, refactored it and changed date format to ISO. Also changed the styling to badge and disabled displaying date by default.

**Pull request merged**